lib/models/addon.js

```diff
@@ -331,7 +332,29 @@ let addonProto = {
   */
   isDevelopingAddon() {
     if (process.env.EMBER_ADDON_ENV === 'development' && (this.parent instanceof Project)) {
-      return this.parent.name() === this.name;
+      const parentName = this.parent.name();
+      // If the name in package.json and index.js match, we're definitely developing
+      if (parentName === this.name) {
+        return true;
+      }
+
+      // If package.json and index.js names do not match, we should check for scoped vs non-scoped
+      const parsed = npa(parentName);
+      let parsedName = parsed.name;
+      if (parsed.scope) {
+        parsedName = parsedName.replace(parsed.scope, '').slice(1);
+      }
+      if (parsedName === this.name) {
+        if (!this._hasWarnedForMismatchedNames) {
+          console.warn(
```

Comment:
Can you swap this for `this.ui.writeWarnLine`? The UI model honors our command line flags like `--silent`..

Reply:
Sure thing, swapping now.